src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -165,8 +165,12 @@ private void runEventLoop(SelectedSelectionKeySet keys)
       int size = tasks.size();
       try {
         if (size > 0) {
+          Runnable task;
           while (alive && size > 0) {
```

Comment:
Do you need the size > 0 in the while as it is dynamically changing, the variable is not being updated and you have an exit condition in the loop.